EpiAware/src/epimodel.jl

```diff
@@ -1,24 +1,24 @@
 abstract type AbstractEpiModel end
 
 
+
 """
-    struct EpiModel{T}
+    struct EpiModel{T<:Real} <: AbstractEpiModel
 
-The `EpiModel` struct represents a basic renewal model.
+EpiModel represents an epidemiological model with generation intervals, delay intervals, and observation delay kernel.
 
 # Fields
-- `gen_int::Vector{T}`: Discrete generation distribution.
-- `delay_int::Vector{T}`: Discrete delay distribution.
-- `delay_kernel::SparseMatrixCSC{T,Integer}`: Sparse matrix representing the action of convoluting
-    a series of infections with observation delay.
-- `cluster_coeff::T`: Cluster coefficient for negative binomial distribution of observations.
-- `len_gen_int::Integer`: Length of `gen_int` vector.
-- `len_delay_int::Integer`: Length of `delay_int` vector.
+- `gen_int::Vector{T}`: Discrete generation inteval, runs from 1, 2, ... to the end of the vector.
+- `delay_int::Vector{T}`: Discrete delay distribution runs from 0, 1, ... to the end of the vector less 1.
+- `delay_kernel::SparseMatrixCSC{T,Integer}`: Sparse matrix representing the observation delay kernel.
+- `cluster_coeff::T`: Cluster coefficient for negative binomial observations.
```

Comment:
can we add some help here for people so they can understand its 1 / overdispersion